### PR TITLE
README: Update build instructions

### LINKED ISSUE #492: README instructions for building flannel are wrong / out-of-date
The README references `./build` which does not exist. As a newcomer to the project, the basic instructions of how to build the tool are incorrect.

### PR DESCRIPTION
**Pull request by tomdee**
Reference the Makefile instead of the old build script

Fixes #492, #480

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,24 +21,12 @@ The following diagram demonstrates the path a packet takes as it traverses the o
 
 ## Building flannel
 
-* Step 1: Make sure you have required dependencies installed on your machine. On Ubuntu, run `sudo apt-get install linux-libc-dev golang gcc`.
-On Fedora/Redhat, run `sudo yum install kernel-headers golang gcc`.
-* Step 2: Git clone the flannel repo: `git clone https://github.com/coreos/flannel.git`
-* Step 3: Run the build script: `cd flannel; ./build`
+To build flannel, run `make dist/flanneld`
 
-### Building in a Docker container
```
Comment (steveeJ) [inline README.md]: Why don't you want to keep a headline for building in a container?

File: README.md [hunk 2]
```diff
@@ -21,24 +21,12 @@ The following diagram demonstrates the path a packet takes as it traverses the o
...
+To build flannel in a container run `make dist/flanneld-amd64` (replace the `amd64` suffix to build on other architectures. See the Makefile for supported architectures).
```
Comment (steveeJ) [inline README.md]: We should mention that Docker is required for this.

File: README.md [hunk 3]
```diff
@@ -21,24 +21,12 @@ The following diagram demonstrates the path a packet takes as it traverses the o
 
 ## Building flannel
 
-* Step 1: Make sure you have required dependencies installed on your machine. On Ubuntu, run `sudo apt-get install linux-libc-dev golang gcc`.
-On Fedora/Redhat, run `sudo yum install kernel-headers golang gcc`.
-* Step 2: Git clone the flannel repo: `git clone https://github.com/coreos/flannel.git`
-* Step 3: Run the build script: `cd flannel; ./build`
+To build flannel, run `make dist/flanneld`
```
Comment (steveeJ) [inline README.md]: What are the requirements for building?

**Comment by tomdee** [comment]
@steveeJ Updated, PTAL.

**Comment by jellonek** [comment]
What about other tools mentioned in `Dockerfile`?

**Comment by tomdee** [comment]
@jellonek What tools do you mean?

**Comment by tomdee** [comment]
@steveeJ I would like to merge this ASAP, it's not great having the bad information sitting on master...

**Comment by jellonek** [comment]
You written that "To build flannel in a container run `make dist/flanneld-amd64`" but tbh container could be build with `make flannel-git` which also requires other tools, `dist/iptables-amd64` and `dist/libpthread.so.0-amd64`.
This seems to me a bit imprecise but still, much better than actual version of docs.

**Pull request merged**